File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        stack = list()

        for letter in s:
            stack.append(letter)
            while (len(stack) > 1 and 
                   (stack[-1] == 'B' and stack[-2] == 'A' or
                    stack[-1] == 'D' and stack[-2] =='C')):
                stack.pop()
                stack.pop()

        return len(stack)

File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.java
class Solution {
    public int minLength(String s) {
        Stack<Character> stack = new Stack();

        for (int i = 0; i < s.length(); i++) {
            stack.add(s.charAt(i));
            while (stack.size() > 1
                    && ((stack.get(stack.size() - 1) == 'B' && stack.get(stack.size() - 2) == 'A')
                            || (stack.get(stack.size() - 1) == 'D'
                                    && stack.get(stack.size() - 2) == 'C'))) {
                stack.pop();
                stack.pop();
            }
        }

        return stack.size();
    }
}
